### DISCUSSION TITLE
Browse messages with Protobuf encoding

### DISCUSSION THREAD

**Discussion by saxo-anlu** [provectus/kafka-ui#2645] [Q&A] [answered]
Hello kafka-ui team,

could you please tell me is it possible to browse messages with Protobuf encoding using kafka-ui. Do I understand correctly that it's possible two ways: SchemaRegistry of via protobuf files (https://github.com/provectus/kafka-ui/blob/master/documentation/guides/Protobuf.md) ?

I've read the documentation and saw this statement "Browse Messages — browse messages with JSON, plain text, and Avro encoding" on the other hand your colleague German Osin in his overview of UI tools for Kakfa wrote that browse messages with Protobuf encoding is supported (https://towardsdatascience.com/overview-of-ui-tools-for-monitoring-and-management-of-apache-kafka-clusters-8c383f897e80). So what is the current status.

We're choosing a tool to monitor our Kafka clusters and kafka-ui is the one for us to go if there is possibility to browse Protobuf messages.

Thank you in advance for your reply.

Kind regards,
Anatol Lutski

**Comment by Haarolean** [comment, marked as answer]
Hey, sorry for the delay, haven't seen this discussion until an old email popped up in my inbox :-)
SerDe plugins setup docs: [here](https://provectus.gitbook.io/kafka-ui/configuration/serialization-serde)
